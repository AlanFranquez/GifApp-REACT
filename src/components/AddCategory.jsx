import { useState } from "react"

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (e) => {
        setInputValue(e.target.value)
        console.log(e.target.value)

    }

    const onSubmit = (e) => {
        e.preventDefault();

        if(inputValue === '') {
            console.log('rellene los datos');

            return;
        }

        // Llamo al arreglo, de esta forma ya que no tengo la primera opciÃ³n de categoria
        setCategories(categories => [inputValue, ...categories]);

        //  Para limpiar el el input
        setInputValue('')
    }

    return (

        <form onSubmit={onSubmit}>
            <input type='text' placeholder='Busca un gif' value={inputValue} onChange={onInputChange}/>
        </form>
        
    )
}
