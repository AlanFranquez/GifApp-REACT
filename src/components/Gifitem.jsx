import Proptypes from 'prop-types'


export const Gifitem = ({title, url}) => {

    // console.log(img.title)
  return (
    <div className="card">
        <p>{title}</p>
        <img src={url} alt={title}></img>
    </div>
  )
}

Gifitem.propTypes = {
  title: Proptypes.string.isRequired,
  url: Proptypes.string.isRequired
}