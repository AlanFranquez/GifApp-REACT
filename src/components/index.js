// Archivo barril para ordenar mejor las exportaciones
export * from './AddCategory'
export * from './GifGrid'
export * from './Gifitem'