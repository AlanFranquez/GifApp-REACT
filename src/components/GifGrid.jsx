import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { Gifitem } from "./Gifitem";
import PropTypes from 'prop-types'


export const GifGrid = ({category}) => {


    // Custom HOOK
    const {images, isLoading} = useFetchGifs(category)

      

    // const [images, setImages] = useState([]);


    // const newImages = async() => {
    //     const insertar = await getGifs(category);
    //     setImages(insertar)
    // }
    
    // useEffect( () => {
    //     newImages();
    // }, []) 
    
    // console.log(images)


    return (
        <>
            <h2>{category}</h2>

            {isLoading ? <p>Cargando...</p> : null}
            {/* Otra forma de hacerlo pero más  corto */}
            {/* {isLoading && <p>Cargando...</p>} */}

            <div className="card-grid">

            {/* Destructuración del img */}
                {images.map((img) => {
                    return <Gifitem key={img.id} {...img} />
                })}

            </div>


        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}