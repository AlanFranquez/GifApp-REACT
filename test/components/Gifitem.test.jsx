import { render, screen } from "@testing-library/react"
import { Gifitem } from "../../src/components/Gifitem"

describe('Testeando el gifitem', () => {

    const title = 'ARISU';
    const url = 'https://google.com/'

    test('Proabndo el proptype', () => {

        const {container} = render(<Gifitem title={title} url={url} />);

        expect(container).toMatchSnapshot()

    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<Gifitem title={title} url={url} />);

        // screen.debug();

        // expect(screen.getByRole("img").src).toBe(url);
        // expect(screen.getByRole("img").alt).toBe(title);

        const {src, alt } = screen.getByRole('img');

        expect(src).toBe(url)
        expect(alt).toBe(alt)

    });

    test('Debe de mostrar el titulo en el componoente', () => {

        render(<Gifitem title={title} url={url} />);


        // Los atributos no los toma como textos, por lo tanto solo aparece uno ac√° 
        expect(screen.getByText(title)).toBeTruthy()

    });

})