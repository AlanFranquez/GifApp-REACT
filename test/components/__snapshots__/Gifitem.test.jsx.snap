// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeando el gifitem Proabndo el proptype 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      ARISU
    </p>
    <img
      alt="ARISU"
      src="https://google.com/"
    />
  </div>
</div>
`;
